docs/vscode-extension.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Deepnote extension for VS Code, Cursor, and Windsurf
+description: Complete guide to the official Deepnote extension for working with Deepnote notebooks locally in VS Code, Cursor, and Windsurf.
+noIndex: false
+noContent: false
+---
+
+# Deepnote extension for VS Code, Cursor, and Windsurf
+
+A powerful extension for [VS Code](https://code.visualstudio.com/), [Cursor](https://cursor.sh/), and [Windsurf](https://windsurf.ai/) that brings Deepnote notebook capabilities directly into your favorite AI-native code editor. Work with slick AI notebooks featuring SQL blocks, database integrations, and reactive blocks.
+
+**Available on:**
+
+- [VS Code Marketplace](https://marketplace.visualstudio.com/items?itemName=deepnote.deepnote)
+- [Open VSX Registry](https://open-vsx.org/extension/deepnote/deepnote) (for Cursor, Windsurf, and other VS Code compatible editors)
+- [GitHub Repository](https://github.com/deepnote/deepnote-vscode)
+
+The Deepnote extension provides:
+
+- **Native `.deepnote` file support** - Open and edit Deepnote notebooks
+- **Block execution** - Run code, SQL, and other executable blocks
+- **Rich outputs** - Display DataFrames, plots, and visualizations
+- **IntelliSense** - Code completion and suggestions
+- **Debugging** - Set breakpoints and debug code blocks
+- **Git integration** - Version control for notebooks
+- **Local execution** - Use your own Python environment
+
+## Installation
+
+### From marketplace
+
+1. **Open your editor**
+   - Launch VS Code, Cursor, or Windsurf
+
+2. **Open Extensions View**
+   - Click Extensions icon in sidebar (Ctrl/Cmd + Shift + X)
+   - Or go to View → Extensions
+
+3. **Search for Deepnote**
+   - Type "Deepnote" in the search box
+   - Find "Deepnote" by Deepnote
+
+4. **Install**
+   - Click "Install" button
+   - Wait for installation to complete
+   - Reload your editor if prompted
+
+### From command line
+
+```bash
+# Install using CLI
+code --install-extension deepnote.deepnote  # VS Code
+cursor --install-extension deepnote.deepnote  # Cursor
+windsurf --install-extension deepnote.deepnote  # Windsurf
+```
+
+### Verify installation
+
+1. Open Command Palette (Ctrl/Cmd + Shift + P)
+2. Type "Deepnote"
+3. You should see Deepnote-related commands
+
+## Features
+
+### Native .deepnote file support
+
+The extension automatically associates `.deepnote` files with the notebook interface, providing YAML syntax highlighting when viewing the raw file and a structured notebook view for editing. You can work in three different view modes: Notebook View for interactive editing and execution, Source View for raw YAML editing, or Split View to see both views side-by-side.
```

Comment:
please clarify how to view the raw .yml file (that's also a feature of vscode not the extension). It's through file explorer -> right click on .deepnote file -> open with -> text editor